src/Debug.c

```diff
@@ -46,6 +47,36 @@ void Ros_Debug_BroadcastMsg(char* fmt, ...)
 
     if (ros_DebugSocket == -1)
         Ros_Debug_Init();
+    
+   // Timestamp
+
+    time_t DEBUG_MSG_TIMESTAMP;
+    struct tm ts;
+    struct timeval tv;
+
+
+    time(&DEBUG_MSG_TIMESTAMP);
+
+    localtime_r(&DEBUG_MSG_TIMESTAMP, &ts);
+
+    strftime(Formatted_Time, sizeof(Formatted_Time), "%a %Y-%m-%d %H:%M:%S", &ts);
+
+    gettimeofday(&tv, NULL);
+    long msec = tv.tv_usec / 1000;
+
+    snprintf(Formatted_Time, sizeof(Formatted_Time), "%s.%03ld", Formatted_Time, msec);
+
+    size_t Timestamp_Length = strlen(Formatted_Time);
+    size_t Debug_Message_Length = strlen(str);
+
+    // Pre-pending the timestamp to the debug message 
+
+    if (Timestamp_Length + Debug_Message_Length + 1 < MAX_DEBUG_MESSAGE_SIZE)
+    {
+        memmove(str + Timestamp_Length, str, Debug_Message_Length + 1); // Move existing contents of str buffer to the end by Timestamp_Length to make space for the timestamp and avoiding overwriting the debug message during the move 
+        memcpy(str, Formatted_Time, Timestamp_Length); // Copy the timestamp stored in Formatted_Timestamp buffer to the beginning of str buffer        
+    }
```

Comment:
Is there anything we can do if the combined string is too long?

Maybe truncate with an ellipsis?

Reply:
Will truncate it with an ellipsis or some other way if that doesn't work.